students/GKim/lesson03/mailroom.py

```diff
@@ -0,0 +1,186 @@
+    new donation amount 
+    """
+    clear_screen()
+    while True:
+        thanks_answer = menu_thank_you().strip()
+        # thanks_answer = thanks_answer.strip()
+        if thanks_answer.upper() == "LIST":
+            clear_screen()
+            show_list()
+        elif thanks_answer.upper() == "MENU":
+            clear_screen()
+            break
+        else:
+            idx = len(main_donors)-1
+            in_main_donors = False
+            for x in range(len(main_donors)-1, -1, -1):
+                if main_donors[x][0].lower() == thanks_answer.lower():
+                    in_main_donors = True
+                    print("\nThe Donor you have selected is {}".format(thanks_answer))
+                    donation_amount = int(input("\nPlease enter the amount that {} kindly donated: ".format(thanks_answer)))
+                    main_donors[x][1].append(donation_amount)
+                    print("{}: ${:.2f}".format(thanks_answer, donation_amount))
+                idx -= 1
+            
+            if in_main_donors == False:
+                donation_amount = int(input("\nPlease enter {}'s donated amount: ".format(thanks_answer)))
+                add_new_donors = (thanks_answer,[donation_amount])
+                main_donors.append(add_new_donors)
+                print("{} was added with a donation of ${:.2f}".format(thanks_answer,donation_amount))
+            
+            send_email(thanks_answer, donation_amount)
+
+
+
+def report_menu():
+    """
+    Menu for Create a Report and to give the user an option to exit
+    """
+    report_menu_answer = input(""" 
+    Welcome to the 'Create a Report option'!
+Enter 'MENU' to exit and return to the Main Menu
+Press Enter to continue
+ >>> \n""")
+
+    return report_menu_answer
+
+
+def gen_stats(main_donors):
+    donor_stats = []
+    lst = main_donors
+    for donor in lst:
+        donations = donor[1]
+        total = sum(donations)
+        num = len(donations)
+        avg = round(total / num, 2)
+        stats = [donor[0], total, num, avg]
+        donor_stats.append(stats)
+
+    return donor_stats
+
+def create_report():
+    """
+    Generates the report of donors by donation amount from greatest to least
+    """
+    while True:
+        response = report_menu().strip()
+        # response = response.strip()
+        if response.upper() == "MENU":
+            break
+        else:
+            stats_list = gen_stats(main_donors)
+            stats_list.sort(key=lambda stats_list: stats_list[1],reverse=True)
+            
+            print("{:<20}|{:^15}|{:^15}|{:>15}".format("Donor Name", "Total Given", "Num Gifts", "Average Gifts"))
+            print("-" * 68)
+            for donor in stats_list:
+                print("{:<20}${:>15} {:>15} ${:>15}".format(donor[0], donor[1], donor[2], donor[3]))
+            print("\nEnd of Report\n")    
+       
+
+def quit():
+    print("You are leaving the Mailroom!")
+    sys.exit()
+
+def mailroom_main():
```

Comment:
Really nice job here George, you went above and beyond with some of the small additions! 

Just a note moving forward, you usually want to remove commented code when making pushes. Otherwise, it can leave your code looking messy and cluttered without a need for it to be